wicketd/src/http_entrypoints.rs

```diff
@@ -328,6 +336,43 @@ async fn get_update_sp(
     Ok(HttpResponseOk(event_report))
 }
 
+/// Resets update state for a sled.
+///
+/// Use this to clear update state after a failed update.
+#[endpoint {
+    method = POST,
+    path = "/clear-update-state/{type}/{slot}",
+}]
+async fn post_clear_update_state(
+    rqctx: RequestContext<ServerContext>,
+    target: Path<SpIdentifier>,
+    opts: TypedBody<ClearUpdateStateOptions>,
+) -> Result<HttpResponseUpdatedNoContent, HttpError> {
+    let log = &rqctx.log;
+    let target = target.into_inner();
+
+    let opts = opts.into_inner();
+    match opts.test_error {
```

Comment:
Could this be a method on `UpdateTestError`? I'm imagining both this and `post_start_update` could do something like

```rust
if let Some(test_error) = opts.test_error.take() {
    return Err(test_error.into_http_error())
}
```

instead of both handling all variants.

Reply:
Done, thanks.